forge-gui/res/cardsfolder/i/imminent_doom.txt

```diff
@@ -6,5 +6,5 @@ T:Mode$ SpellCast | ValidCard$ Card.cmcEQX | ValidActivatingPlayer$ You | Trigge
 SVar:TrigDealDamage:DB$ DealDamage | ValidTgts$ Any | NumDmg$ Y | SubAbility$ DBPutCounter
 SVar:DBPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ DOOM | CounterNum$ 1
 SVar:X:Count$CardCounters.DOOM
-SVar:Y:LastStateBattlefield$CardCounters.DOOM
+SVar:Y:ThisTurnCast$Card.TriggeredCard$CardManaCost
```

Comment:
Can't you get the cmc from the Triggered Spell or something?
`SVar:Y:TriggeredStackInstance$CardManaCostLKI`
There is no need for extra This Turn Cast?

Reply:
@tool4ever ☝️

Reply:
@Hanmac 
not really I think
this card needs to always look back:
- so if I use `TriggeredSpellAbility`
- it's X spell
- there is _Unbound Flourishing_

it might deal the wrong damage :/

Reply:
> The amount of damage Imminent Doom’s triggered ability deals is the number of counters it had on it as the ability triggered.

Reply:
> > The amount of damage Imminent Doom’s triggered ability deals is the number of counters it had on it as the ability triggered.

But that amount should be cmc when the ability triggered

Reply:
- yes and if I cast _Endless One_ with X=1
- then resolve _Unbound Flourishing_ first
- `TriggeredStackInstance$CardManaCostLKI` is 2

I wrote it explicitly to only use LKI if the zone changed because other cards like _Vial Smasher_ should see the current value if X gets changed by that special card

Reply:
yeah, its CastSACMC that got removed with this:
https://github.com/Card-Forge/forge/commit/d011c219b9dc142fd56aaf060cd764e74c566cca

Trigger sets these three values:
```java
        sa.setTriggeringObject(AbilityKey.Card, castSA.getHostCard());
        sa.setTriggeringObject(AbilityKey.SpellAbility, castSA.copy(castSA.getHostCard(), true));
        sa.setTriggeringObject(AbilityKey.StackInstance, si);
```

Then Unbound Flourishing updates X for The HostCard and the StackInstance?
https://github.com/Card-Forge/forge/blob/master/forge-game/src/main/java/forge/game/ability/effects/ChangeXEffect.java

so: `AbilityKey.SpellAbility` shouldn't be touched, but `AbilityKey.Card` gets updated, right?

maybe that should be an LKI object too?